doc/decisions/1_in_discussion/spec_expressiveness.md

```diff
@@ -2,12 +2,38 @@
 
 ## Problem
 
-Currently, you can easily come into wrong assumptions that something would work in the specification.
-
-We need to find minimal requirements to implement a sane spec plugin.
+The current implementation of the spec plugin is copying metadata from the specification
+(`spec` namespace) to all other namespaces. By providing certain metadata e.g. 
+`type` or `description`, it can be assured that keys adhere to a given specification. 
+
+For some applications there are keys we don't want to define for every application explicitly.
+This is when `globbing` expressions in key names comes into place. By providing certain wildcard
+characters e.g. `_` or `#` one can match and specify a number of keys at once without the need
+to explicitly state the key in every specification.
+
+To describe the use case where `globbing` expressions make sense, we will use an application
+based on microservice architecture as sample.
```

Comment:
Please describe use-cases separately (in another PR). For web-UI there is already doc/usecases/elektra_web/UC_key_validation.md

We also need use-cases directly for the spec lang, e.g., for `#` and `_` (we probably don't need `*`).